vyper/semantics/analysis/local.py

```diff
@@ -778,10 +771,11 @@ def _analyse_range_call(node: vy_ast.Call) -> list[VyperType]:
 
     if "bound" in kwargs:
         bound = kwargs["bound"]
-        folded_bound = bound.get_folded_value() if bound.has_folded_value else bound
-        if not isinstance(folded_bound, vy_ast.Num):
+        if bound.has_folded_value:
+            bound = bound.get_folded_value()
+        if not isinstance(bound, vy_ast.Num):
```

Comment:
just use try/catch here - if we get an UnfoldableNode, re-raise with StateAccessViolation